independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanInfo.java

```diff
@@ -364,6 +365,61 @@ void validate(List<Throwable> errors, List<BeanDeploymentValidator> validators)
                                         "%s bean is not proxyable because it has a private no-args constructor: %s. To fix this problem, change the constructor to be package-private",
                                         classifier, this)));
             }
+        } else if (isProducerMethod()) {
+            MethodInfo methodInfo = target.get().asMethod();
```

Comment:
I believe that the code for producer methods and fields could be shared?

Reply:
Good point